src/systems/ec/implementations/vanilla/packages/ec-component-site-switcher/ec-component-site-switcher.scss

```diff
@@ -5,10 +5,110 @@
 
 // Import base and generic
 @import '@ecl/ec-base/ec-base';
-@import '@ecl/generic-component-site-switcher/generic-component-site-switcher';
 
 // Check if overridden dependencies are already loaded, if needed
-@include check-imports(('ec-layout-grid', 'ec-component-link'));
+@include check-imports(('ec-layout-grid'));
+
+@mixin ecl-site-switcher() {
+  .ecl-site-switcher {
+    @include ecl-hidden-print();
+
+    background-color: $ecl-color-grey-5;
+    display: block;
+    font: $ecl-font-m;
+    margin: 0;
+
+    &::after {
+      clear: both;
+      content: '';
+      display: block;
+    }
+  }
+
+  .ecl-site-switcher__link {
+    display: block;
+    font-weight: bold;
+    line-height: 1.3;
+    padding: $ecl-spacing-m $ecl-spacing-m;
+    text-decoration: none;
+
+    &::after,
+    &--is-external::after {
+      content: none !important;
+    }
+
+    &:hover,
+    &:focus {
+      outline-offset: -3px;
+      text-decoration: underline;
+    }
+  }
+
+  .ecl-site-switcher__list {
+    display: flex;
+    list-style: none;
+    margin: 0;
+    padding: 0;
+  }
+
+  .ecl-site-switcher__option {
+    display: inline-block;
+    flex-grow: 1;
+    text-align: center;
+    vertical-align: middle;
+  }
+
+  .ecl-site-switcher__option--is-selected {
+    background-color: $ecl-color-primary;
+
+    .ecl-site-switcher__link {
+      color: #fff;
+    }
+  }
+
+  .ecl-site-switcher--footer .ecl-site-switcher__option,
+  .ecl-site-switcher--header .ecl-site-switcher__option {
+    position: relative;
+
+    &--is-selected::after {
+      border: 10px solid rgba(0, 68, 148, 0);
+      content: ' ' !important;
+      height: 0;
+      left: 50%;
+      margin-left: -10px;
+      pointer-events: none;
+      position: absolute;
+      width: 0;
+    }
+  }
+
+  .ecl-site-switcher--header .ecl-site-switcher__option {
+    &--is-selected::after {
+      border-top-color: $ecl-color-primary;
+      top: 95%;
+    }
+  }
+
+  .ecl-site-switcher--footer .ecl-site-switcher__option {
+    &--is-selected::after {
+      border-bottom-color: $ecl-color-primary;
+      bottom: 95%;
+    }
+  }
+
+  /* stylelint-disable-next-line order/order */
+  @include ecl-media-breakpoint-up(md) {
+    .ecl-site-switcher__option {
+      flex-grow: initial;
+    }
+
+    .ecl-site-switcher--footer {
+      .ecl-site-switcher__list {
+        float: right;
```

Comment:
why not use `justify-content: flex-end;`? then no more floats -> no more `clear: both` after -> cleaner